### DISCUSSION TITLE
Preparing Input Data for mhm Model

### DISCUSSION THREAD

**Discussion by soilm-dev** [mhm-ufz/mHM#100] [Q&A] [answered]
Hello All,

I am relatively very much new to mhm model and would like to explore its usage.
I had gone through the manual and found it very much intriguing and at the same time, felt it was not made to be understandable easily. I was able to understand half of it and had prepared majority of the data. However I have the following queries which needed inputs from the developers or the users.

1) Regarding the projection system of all the input data: I have NetCDF files of the morphological data projected in WGS-84,  whereas my morphological data is of UTM projection. I sense this inconsistency is not to be given to the model. Should I reproject all to WGS-84 or to UTM. Coming to the same, How can I reproject a NetCDF file to UTM as I've scrounged all the sources I could get, but was not able to do it. Could anyone guide me in NetCDF processing for mhm model ? It would be helpful if I get any input in preparing the netcdf file (apart from clipping spatially and temporally). 

2) I saw that there are four resolution in mhm model namely L0, L1, L11 and L2. I have fixed my L0 as 100m owing to my DEM data as 90m (as suggested in the manual). L2 resolution of my meteorological files are of 0.25 degree resolution (in degrees). I as a user am in control for the L0 and L2 resolution right?, Now how do I fix the L1, L11 resolution ?

3) LAI class file, my understanding of LAI input is to either give it as a class file or as .nc file having grids of mean monthly average with 12 time steps (for each month). I'm not really sure on how to prepare the LAI_class.asc file as on what basis does the classification be done?, what will be the class for built-up, waterbody? From which data source can I get the mean monthly average? how to create the look-up table file? How do I create regions in my study area and assign those classes? Given the confusion I had, I tried resorting to input a LAI netcdf file. However I faced problem in regridding and reprojecting it. I had downloaded the data with 0.25 degree resolution which wont match with the L0 resolution of 100m. How do I process this netcdf LAI data for mhm?.
I am also curious on the ways to create LAI_class.asc file as I did not get any source or resource for the same.

I sense that all my issues are with either the projection or resolution of the data. 

4) I am also curious on why does this model require all the data in a single resolution? Is there any significance for it?

I had gone through the mhm-tools documentation also, where I got lost and was not able to understand the usage of it. I feel this could be a reason because of my beginner status. 

It would be really helpful for me to proceed as I'm really stuck at this point with all soirces searched and all discussions in this forum read. 

Thanking you in advance!

All the best.

**Comment by StephanThober** [comment, marked as answer]
Dear soilm_dev,

1.) Which kind of domain do you plan to run mHM on? It is important that all the input data is on the same projection. If all your data is in the same UTM zone, then it should be possible to use it as input for mHM. However, if your data spans multiple UTM zones, then you should consider an alternative projection. I have never projected data to/from UTM, but tools like gdal (https://gdal.org/en/latest/) might be helpful.

2.) You are right. In general, the user can choose L1 and L11, however, these need to be multiples of L0 and fractions/multiples of L2. Commonly, L1 and L11 are chosen to be equal to L2.

3.) The question here is what is the underlying land cover information you are using? The LAI_class.asc contains the ids of the land cover class present at a given L0 grid cell. In the look-up table, contained in the LAI_classdefinition.txt, the mean LAI values for a given month and land cover class are noted down. You find examples here: https://git.ufz.de/mhm/mhm/-/tree/develop/test_domain/input/morph

4.) Yes, all morphological information should be provided on L0. I am wondering which dataset you used for LAI at 0.25 degree. Most land cover maps and satelite information like MODIS provide LAI on a substantially finer grid.

I hope this helps.

Best,
Stephan
